datadog_checks_base/datadog_checks/base/stubs/_util.py

```diff
@@ -20,7 +20,7 @@ def subprocess_output(command, raise_on_empty_output):
     # docs warn that the data read is buffered in memory. They suggest not to
     # use subprocess.PIPE if the data size is large or unlimited.
     with tempfile.TemporaryFile() as stdout_f, tempfile.TemporaryFile() as stderr_f:
-        proc = subprocess.Popen(command, stdout=stdout_f, stderr=stderr_f)
+        proc = subprocess.Popen(command, stdout=stdout_f, stderr=stderr_f, env=env)
```

Comment:
note: this implementation of `subprocess_output` is a stub when this doesn't run within the python interpreter embedded in the Agent. In an Agent, this function is implemented here: https://github.com/DataDog/datadog-agent/blob/e291c6b0c954a19c2f12e67da80cd91a5ce85f29/pkg/collector/python/util.go#L28, and would need to be updated to support passing an env.

Reply:
Here is the PR to add the support of the `env` parameter to `_util.subprocess_output`: DataDog/datadog-agent#5987